packages/integration-tests/src/api/session.ts

```diff
@@ -51,112 +51,148 @@ export const consent = async (interactionCookie: string) =>
     .json<RedirectResponse>();
 
 export const sendRegisterUserWithEmailPasscode = (email: string, interactionCookie: string) =>
-  api.post('session/register/passwordless/email/send-passcode', {
+  api.post('session/passwordless/email/send', {
     headers: {
       cookie: interactionCookie,
     },
     json: {
       email,
+      flow: 'register',
     },
   });
 
-export const verifyRegisterUserWithEmailPasscode = (
+export const verifyRegisterUserWithEmailPasscode = async (
   email: string,
   code: string,
   interactionCookie: string
-) =>
-  api
-    .post('session/register/passwordless/email/verify-passcode', {
+) => {
+  await api.post('session/passwordless/email/verify', {
+    headers: {
+      cookie: interactionCookie,
+    },
+    json: {
+      email,
+      code,
+      flow: 'register',
+    },
+  });
+
+  return api
+    .post('session/register/passwordless/email', {
       headers: {
         cookie: interactionCookie,
       },
-      json: {
-        email,
-        code,
-      },
     })
     .json<RedirectResponse>();
+};
```

Comment:
sinse this is a api integration test, better split these into two steps. verify and register.